torch/distributed/distributed_c10d.py

```diff
@@ -4297,6 +4298,175 @@ def _get_backend_from_str(backend: Optional[str] = None) -> Backend:
     return Backend(backend)
 
 
+@_time_logger
+def split_group(
+    parent_pg=None,
+    split_ranks=None,
+    timeout=None,
+    pg_options=None,
+    group_desc=None,
+):
+    """
+    Create a new distributed group splitted from the given parent group.
+
+    users of this API must gurantee that all ranks in the parent group enter this API call.
+    And the split of the group is the same accross the ranks.
+
+    Args:
+        parent_pg (ProcessGroup, optional): The parent process group. If None,
+            the default process group will be used. Users need to gurantee that 
+            the parent group is fully initialized (e.g, communicators are initialized)
+        split_ranks (list[int]): the split ranks, which is a list of list of ranks.
```

Comment:
typing looks wrong? you say list[int] and then later you say list of list of int.  

also, i'm reading this and not understanding how split_ranks actually works.  need to say what each int represents and if there are nested lists, what each list represents.